### PR TITLE
OT166-45: Endpoint Detalle Novedad

### PR DESCRIPTION
**Pull request by devfacu**
# [Ticket OT166-45](https://alkemy-labs.atlassian.net/browse/OT166-45)

* Main branch merged and NewsDetailService created.
* NewsDetailResponse POJO created.
* IGetNewsDetailService interface created and renamed.
* Renamed NewsDetailService and renamed again for not paying attention.
* getBy method created in NewsResource.
* IGetNewsDetailService name updated.
* toNewsDetailResponse method added to INewsMapper.
* getBy method added in NewsService, IGetNewsDetailService implemented.
* filter for "GET news/id" endpoint added.
* Postman collection updated.

### HOW HAS THIS BEEN TESTED?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration.

- [x] Postman.
- [ ] Unit test.
- [ ] Integration test.
- [ ] No testing required (only applies for tech task).

**Test Configuration**:

* Valid ID
* No valid ID

**Screenshots**:

* TestGetNewsDetail

* TestGetNewsDetail 2

### CHECKLIST

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/application/rest/response/NewsDetailResponse.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@NoArgsConstructor
+@Getter
+@Setter
+public class NewsDetailResponse {
+
+  private Long id;
+  private String name;
+  private String text;
+  private String image;
+  private CategoryResponse category;
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/application/rest/response/NewsDetailResponse.java:18]: 💯

File: src/main/java/com/alkemy/ong/application/rest/response/NewsDetailResponse.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+public class NewsDetailResponse {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/application/rest/response/NewsDetailResponse.java:12]: Reutilizar NewsResponse con las annotations correspondientes.

File: src/main/java/com/alkemy/ong/infrastructure/database/mapper/abstraction/INewsMapper.java [hunk 3]
```diff
@@ -16,4 +17,7 @@
   @Mappings({@Mapping(target = "text", source = "newsEntity.content")})
   NewsResponse toNewsResponse(NewsEntity newsEntity);
 
+  @Mappings({@Mapping(target = "text", source = "newsEntity.content")})
+  NewsDetailResponse toNewsDetailResponse(NewsEntity newsEntity);
+
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/mapper/abstraction/INewsMapper.java:22]: y la category?

**Review by mkain0 — CHANGES_REQUESTED** [review]

**Comment by mkain0** [comment]
@devfacu Por favor sync con main y reutilizar las interfaces creadas por @MatiasFiorentini

File: src/main/java/com/alkemy/ong/infrastructure/database/mapper/abstraction/INewsMapper.java (hunk 3 above)
Comment (devfacu) [reply src/main/java/com/alkemy/ong/infrastructure/database/mapper/abstraction/INewsMapper.java:22]: La category la levanta automáticamente como se puede ver en la response del print de pantalla. Si no es eso entonces no entiendo el comentario jeje.

Comment (mkain0) [reply src/main/java/com/alkemy/ong/infrastructure/database/mapper/abstraction/INewsMapper.java:22]: Ahhh... si, porque en este caso es un object. En otro caso lo hicimos un field.

**Review by mkain0 — APPROVED** [review]

**Pull request merged**